src/integer_mod_q/mat_polynomial_ring_zq/properties.rs

```diff
@@ -0,0 +1,202 @@
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, PolyOverZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = PolyOverZq::from_str("5  1 0 0 0 1 mod 17").unwrap();
+    /// let id_mat = MatPolyOverZ::identity(2, 2);
+    ///
+    /// let poly_ring_mat = MatPolynomialRingZq::from((id_mat, modulus));
+    /// assert!(poly_ring_mat.is_identity());
+    /// ```
+    ///
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, PolyOverZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = PolyOverZq::from_str("5  1 0 0 0 1 mod 17").unwrap();
+    /// let id_mat = MatPolyOverZ::from_str("[[1  1, 0],[0, 1  1],[0, 0]]").unwrap();
+    ///
+    /// let poly_ring_mat = MatPolynomialRingZq::from((id_mat, modulus));
+    /// assert!(poly_ring_mat.is_identity());
+    /// ```
+    pub fn is_identity(&self) -> bool {
+        self.matrix.is_identity()
+    }
+
+    /// Checks if a [`MatPolynomialRingZq`] is a square matrix.
+    ///
+    /// Returns true if the number of rows and columns is identical.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, PolyOverZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = PolyOverZq::from_str("5  1 0 0 0 1 mod 17").unwrap();
+    /// let poly_mat = MatPolyOverZ::from_str("[[1  13, 0],[2  1 1, 1  1]]").unwrap();
+    ///
+    /// let poly_ring_mat = MatPolynomialRingZq::from((poly_mat, modulus));
+    /// assert!(poly_ring_mat.is_square());
+    /// ```
+    pub fn is_square(&self) -> bool {
+        self.matrix.is_square()
+    }
+
+    /// Checks if every entry of a [`MatPolynomialRingZq`] is `0`.
+    ///
+    /// Returns true if every entry is `0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, PolyOverZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = PolyOverZq::from_str("5  1 0 0 0 1 mod 17").unwrap();
+    /// let poly_mat = MatPolyOverZ::from_str("[[0, 0],[0, 0]]").unwrap();
+    ///
+    /// let poly_ring_mat = MatPolynomialRingZq::from((poly_mat, modulus));
+    /// assert!(poly_ring_mat.is_zero());
+    /// ```
+    pub fn is_zero(&self) -> bool {
+        self.matrix.is_zero()
+    }
+}
+
+#[cfg(test)]
+mod test_is_identity {
+    use crate::{
+        integer::MatPolyOverZ,
+        integer_mod_q::{MatPolynomialRingZq, PolyOverZq},
+    };
+    use std::str::FromStr;
+
+    /// Ensure that is_identity returns `true` for identity matrices.
+    #[test]
+    fn identity_detection() {
+        let modulus = PolyOverZq::from_str("5  1 0 0 0 1 mod 17").unwrap();
+        let poly_mat = MatPolyOverZ::from_str("[[1  1, 0],[0, 1  1],[0, 0]]").unwrap();
```

Comment:
would prefer this tbh
```suggestion
        let poly_mat = MatPolyOverZ::identity(2,3);
```